Cargo.toml

```diff
@@ -68,6 +68,7 @@ script = ["sha1_smol"]
 tls = ["native-tls"]
 async-std-comp = ["aio", "async-std"]
 async-std-tls-comp = ["async-std-comp", "async-native-tls", "tls"]
+testing = ["futures"]
```

Comment:
Is this intended to be part of the public API?

Reply:
Yes, it is intended to be part of the public API so that users (like me) can incorporate these test helpers into the unit tests in our codebases. I have made it be a feature here (which would be enabled as a dev dependency), but it could also be put into a separate crate, much like Tokio has done with the [`tokio-test` crate](https://github.com/tokio-rs/tokio/tree/master/tokio-test). Thoughts?

Reply:
A separate crate would allow evolving the API separately from the rest of the crate's API. I suppose we could keep the feature unstable for now (as in, we don't guarantee the public API to be compatible across semver-compatible versions of the crate).

Reply:
I will restructure this PR to create a `redis-test` crate. Is that acceptable?

Reply:
@jaymell what do you think? I'm just worried the API might be unstable, but maybe it's not so bad?

Reply:
Hmm, I would say that a separate crate certainly does sound like an ideal solution, though the additional maintenance costs there do temper my enthusiasm somewhat :sweat_smile:

Reply:
I'll switch to separate crate then (but the code will still live in this repo).